modules/environment/default.nix

```diff
@@ -193,6 +198,21 @@ in
       ${cfg.extraInit}
     '';
 
+    system.build.setEnvironmentFish = pkgs.writeText "set-environment-fish" ''
+      # Prevent this file from being sourced by child shells.
+      set -g -x __NIX_DARWIN_SET_ENVIRONMENT_DONE 1
+
+      set -g -x PATH ${config.environment.systemPath}
+      ${concatStringsSep "\n" exportVariablesFish}
+
+      # Extra initialisation
+      set fish_function_path ${pkgs.fish-foreign-env}/share/fish-foreign-env/functions $__fish_datadir/functions
```

Comment:
```suggestion
      set fish_function_path ${pkgs.fish-foreign-env}/share/fish-foreign-env/vendor_functions.d $__fish_datadir/functions
```

Reply:
Thank you! Happy to commit this suggestion. Would you mind expanding on why you selected this convention? I've definitely seen it before, but would like to hear about how it's typically used.